src/main/docs/guide/config/propertySource.adoc

```diff
@@ -21,6 +21,12 @@ Micronaut by default contains `PropertySourceLoader` implementations that load p
 
 TIP: `.properties`, `.json`, `.yml` are supported out of the box. For Groovy users `.groovy` is supported as well.
 
+=== Secrets and Sensitive Configuration
+
+It is important to note that it is not recommended to store sensitive configuration such as passwords and tokens within configuration files that can potentially be checked into source control systems. 
+
+It is good practise to instead externalize sensitive configuration completely from the application code using preferably a external secret manager system (there are many options here, many provided by Cloud providers) or environment variables that are set during the deployment of the application.
```

Comment:
It would be good to link to our integrations with Secrets Manager solutions. `[AWS Secrets Manager](https://micronaut-projects.github.io/micronaut-aws/latest/guide/#secretsmanager), [Google Cloud Secrets Manager](https://micronaut-projects.github.io/micronaut-gcp/latest/guide/#secretManager), [HashiCorp Vault](https://docs.micronaut.io/latest/guide/#distributedConfigurationVault)`